packages/amplify_api/ios/Classes/SwiftAmplifyApiPlugin.swift

```diff
@@ -51,13 +46,37 @@ public class SwiftAmplifyApiPlugin: NSObject, FlutterPlugin {
     
     // Create separate method to allow unit testing as we cannot mock "FlutterMethodCall"
     public func innerHandle(method: String, callArgs: Any, result: @escaping FlutterResult){
-        
         do {
             if(method == "cancel"){
                 let cancelToken = try FlutterApiRequest.getCancelToken(args: callArgs)
                 onCancel(flutterResult: result, cancelToken: cancelToken)
                 return
             }
+            else if(method == "addPlugin"){
+                do {
+                    try Amplify.add(plugin: AWSAPIPlugin() )
+                    result(true)
```

Comment:
> Can we add unit tests?

The methods added seem rather "shallow" in the sense that we are pretty much calling an addPlugin method or throwing an exception.  Thus, I imagine a unit test would check that a mocked addPlugin method call returns success, when we say it should return success, or that a mocked addPlugin method would throw an exception when we tell it to throw that same exception.  I imagine such a unit test wouldn't add much value in terms of testing - what are your thoughts on that?

Reply:
You're referring to AmplifyAnalyticsBridge in Android right?  The Android code for Auth, Analytics, API uses result.success(null) while in iOS they use result(true).  

I believe this is because the result object is handled differently between platforms. 

In Android we call result.success or result.failure result.error

In iOS we call result(FlutterError...) or result(info).  I don't believe there's an actual result.success or result.failure in iOS right?  Therefore I believe result.success(null) is for Android and result(true) is for iOS right? 

However, I think you have a point for result(false).  I shouldn't be calling that, instead I should be passing in an exception as result(false) means we're hiding the error (we don't look for boolean returned on dart side)

Reply:
Not so much wondering about the platforms, but the difference between categories.

In `AmplifyApiPlugin.kt ` we do `result(true)`, but in `AmplifyAnalyticsBridge.kt ` we do `flutterResult.success(null)`. These are the same methods being implemented all at once on the same platform, but different categories.  We is there a difference in what they are returning?

Reply:
Hmm I'm not too sure what you're referring to then.  

I did a global search for result(true) and only see it called in swift files.  
Similarly I did a search for flutterResult.success(null) and only see it called in kotlin files.